### PR TITLE
Fix waterlogging checking affected location

### LINKED ISSUE #1155: Water in the fence (with video)
What happened:
It's possible to pour out the water in the fence and this water is flowing in the claim. I send you a video below how this work. Water-flow protect doesn't work on this because it's in the claim, but you can pour out the water even if you're not "trusted person". It's probably link to the water mechanics from 1.13

Steps to reproduce:
https://www.youtube.com/watch?v=X62SDcS31dU

1. Pour out the water in the fence on the plot border (when you're not trusted person)
2. The water is flowing in the claim (you're not trusted so you shouldn't be able to water-flow in the plot)
3.No profit
Server and GriefPrevention version:
The newest version of GriefPrevention 16.16.0
Server version is the newest Paper 1.16.4

Stack trace/error or server log
No console error

### GriefPrevention config.yml
config.yml not affected

### Plugin list (if applicable):
without affecting the bug

### PR DESCRIPTION
**Pull request by nouish**
For blocks that may be waterlogged, checking the relative adjacent location, is wrong. Currently non-trustees may waterlog certain blocks placed on the claim boundary, from outside said claim.

This does not affect waterloggable blocks with containers (chests).

Fixes #1155

### CODE REVIEW

**Review by Jikoo — CHANGES_REQUESTED** [review]
Why add a manually maintained material list when [Waterlogged](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/block/data/Waterlogged.html) exists?

**Comment by nouish** [comment]
> Why add a manually maintained material list when [Waterlogged](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/block/data/Waterlogged.html) exists?

Unfamiliar with the Bukkit/Spigot API, and because not *all* waterlogged blocks are affected. That said, it shouldn't matter if the unaffected ones are included. Will verify.

**Comment by Jikoo** [comment]
I'd guess it would actually create an edge case allowing waterlogging by users with ContainerTrust who should not be able to build for those objects.

I suggest something like this (not accurate, sorry, mobile):
```Java
Block block = event.getBlock();
if (event.getItem() != Material.WATER_BUCKET || !(block.getBlockData() instanceof Waterlogged))
{
  block = block.getRelative(event.getFace());
}
```

Is it possible for users without trust to also remove water from waterlogged blocks?

**Comment by nouish** [comment]
Excellent suggestion!

That also lets me remove that caveat of having to add the handful of waterlogged blocks comming in 1.17 at later date.

> I'd guess it would actually create an edge case allowing waterlogging by users with ContainerTrust who should not be able to build for those objects.

Fortunately not :smile: It will never use CT checks for this!

**Comment by Jikoo** [comment]
> Fortunately not 😄 It will never use CT checks for this!

I meant with the manually specified materials not including containers, since GP was likely only blocking the water placement because it thought the player was attempting to access the inventory.

Looks good!

**Review by Jikoo — APPROVED** [review]
Guess I have to review again to dismiss a review, weird. Any rate, need Robo's final approval on this, but you have mine.

**Review by RoboMWM — APPROVED** [review]

**Pull request merged**
